python/pyspark/sql/session.py

```diff
@@ -2002,6 +2003,61 @@ def addArtifacts(
 
     addArtifact = addArtifacts
 
+    def registerProgressHandler(self, handler: "ProgressHandler") -> None:
+        """
+        Register a progress handler to be called when a progress update is received from the server.
+
+        .. versionadded:: 4.0
```

Comment:
Potentially silly question, but: When you look at [the docs for this][1], it's not obvious that Spark Connect supports this method. Should this be explicitly noted in the docstring somehow? Or are users supposed to assume that everything supports Spark Connect unless explicitly noted otherwise?

[1]: https://spark.apache.org/docs/4.0.0/api/python/reference/pyspark.sql/api/pyspark.sql.SparkSession.registerProgressHandler.html?highlight=progress#pyspark.sql.SparkSession.registerProgressHandler

Reply:
Another related question: Should there be narrative documentation of `ProgressHandler` on the [monitoring page][1], or are we happy with it just being tucked away in the API docs?

[1]: https://spark.apache.org/docs/4.0.0/monitoring.html

Reply:
Creating a PR with documentation updates would be very much appreciated!